src/durbin_watson.jl

```diff
@@ -0,0 +1,238 @@
+
+Default is a two-sided p-value for the alternative hypothesis of positive or negative
+serial correlation. One-sided p-values can be requested by calling
+`pvalue(x::DurbinWatsonTest; tail=)` with the options `:left` (negative serial correlation)
+and `:right` (positive serial correlation).
+
+# References
+
+  * J. Durbin and G. S. Watson, 1951, "Testing for Serial Correlation in Least Squares
+  Regression: II", Biometrika, Vol. 38, No. 1/2, pp. 159-177,
+  [http://www.jstor.org/stable/2332325](http://www.jstor.org/stable/2332325).
+  * J. Durbin and G. S. Watson, 1950, "Testing for Serial Correlation in Least Squares
+  Regression: I", Biometrika, Vol. 37, No. 3/4, pp. 409-428,
+  [http://www.jstor.org/stable/2332391](http://www.jstor.org/stable/2332391).
+  * R. W. Farebrother, 1980, "Algorithm AS 153: Pan's Procedure for the Tail Probabilities
+  of the Durbin-Watson Statistic", Journal of the Royal Statistical Society, Series C
+  (Applied Statistics), Vol. 29, No. 2, pp. 224-227,
+  [http://www.jstor.org/stable/2986316](http://www.jstor.org/stable/2986316).
+
+# External links
+
+  * [Durbin-Watson test on Wikipedia:
+  https://en.wikipedia.org/wiki/Durbin–Watson_statistic
+  ](https://en.wikipedia.org/wiki/Durbin–Watson_statistic)
+"""
+function DurbinWatsonTest{T<:Real}(xmat::AbstractArray{T}, e::AbstractArray{T};
+    p_compute::Symbol = :ndep)
+
+    n = length(e)
+    DW = sum(diff(e) .^2) / sum(e .^2)
+
+    DurbinWatsonTest(xmat, n, DW, p_compute)
+end
+
+testname(::DurbinWatsonTest) = "Durbin-Watson autocorrelation test"
+population_param_of_interest(x::DurbinWatsonTest) =
+    ("sample autocorrelation parameter", "0", 1 - x.DW / 2)
+default_tail(test::DurbinWatsonTest) = :both
+
+function show_params(io::IO, x::DurbinWatsonTest, ident)
+    println(io, ident, "number of observations:     ", x.n)
+    println(io, ident, "DW statistic:               ", x.DW)
+end
+
+"""
+    pan_algorithm(a::AbstractArray, x::Float64, m::Int, n::Int)
+
+Compute exact p-values for the Durbin-Watson statistic using Pan's algorithm (Farebrother,
+1980).
+
+`a` is the vector of non-zero Eigenvalues of ``(I-(X(X'X)^{-1}X'))A`` (see Durbin and
+Watson, 1971, p. 2), `x` is the value of the Durbin-Watson statistic, `m` the number of
+elements in `a`, and `n` the number of approximation terms (see Farebrother, 1980, eq. 5).
+
+# References
+
+  * J. Durbin and G. S. Watson, 1971, "Testing for Serial Correlation in Least Squares
+  Regression: III", Biometrika, Vol. 58, No. 1, pp. 1-19,
+  [http://www.jstor.org/stable/2334313](http://www.jstor.org/stable/2334313).
+  * R. W. Farebrother, 1980, "Algorithm AS 153: Pan's Procedure for the Tail Probabilities
+  of the Durbin-Watson Statistic", Journal of the Royal Statistical Society, Series C
+  (Applied Statistics), Vol. 29, No. 2, pp. 224-227,
+  [http://www.jstor.org/stable/2986316](http://www.jstor.org/stable/2986316).
+
+"""
+function pan_algorithm(a::AbstractArray, x::Float64, m::Int, n::Int)
+
```

Comment:
The convention used in Julia and in JuliaStats packages is not to add blank lines at the top nor at the end of functions.